common/process/src/process_args.cc

```diff
@@ -0,0 +1,179 @@
+/**
+ * Copyright 2011-2013 Centreon
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ */
+
+#include "com/centreon/common/process/process_args.hh"
+#include <stdexcept>
+
+using namespace com::centreon::common;
+using com::centreon::exceptions::msg_fmt;
+
+/**
+ * @brief Construct a new process args::process args object
+ *
+ * @param exe_path first field of cmdline
+ * @param args following arguments
+ */
+process_args::process_args(const std::string_view& exe_path,
+                           std::vector<std::string>&& args)
+    : _exe_path(exe_path), _args(args) {
+  _c_args.reserve(_args.size() + 2);
+  _c_args.push_back(_exe_path.c_str());
+  for (const std::string& arg : _args) {
+    _c_args.push_back(arg.c_str());
+  }
+  _c_args.push_back(nullptr);
+}
+
+/**
+ * @brief Construct a new process args::process args object with an unix style
+ * commandline
+ *
+ * Caution, as this is unix case, get_args() will return an empty vector, only
+ * get_c_args() will contains fields
```

Comment:
will contain fields.

Reply:
There is globally three fields:
* _buffer
* _c_args
* _args

Could you explain a little more in the header file when they are set / not set.